src/hooks/useSafeTimeout.ts

```diff
@@ -0,0 +1,33 @@
+import {useCallback, useEffect, useRef} from 'react'
+
+type SetTimeout = (handler: TimerHandler, timeout?: number, ...args: any[]) => number
```

Comment:
Same suggestion about using `typeof window.setTimeout` as above.

Reply:
I tried that, but there are some required properties on the native type that doesn't make sense to pass in manually to these wrapper functions:

```
Property '__promisify__' is missing in type '(handler: TimerHandler, timeout?: number | undefined, ...args: any[]) => number' but required in type 'typeof setTimeout'.ts(2322)

````

Reply:
Oh weird! Thanks for checking.